kernel/component-manager/src/main/java/org/sakaiproject/modi/SharedApplicationContext.java

```diff
@@ -0,0 +1,70 @@
+package org.sakaiproject.modi;
+
+import org.sakaiproject.util.BeanFactoryPostProcessorCreator;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.AbstractRefreshableApplicationContext;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * An {@link ApplicationContext} that can be shared across webapps as a parent. This serves as the main container for
+ * Sakai, and the effective replacement for the {@link org.sakaiproject.component.api.ComponentManager}.
+ * <p>
+ * It loads beans indirectly, from sources that can yield definitions. This inversion allows components (or overrides,
+ * or any source of beans) to be read or prepared in any format, rather than requiring XML files in a certain layout.
+ */
+public class SharedApplicationContext extends AbstractRefreshableApplicationContext {
```

Comment:
I would like to propose the SharedApplicationContext be renamed to SakaiApplicationContext.

I know this is trivial but when discussing (training) it's helpful to refer to it in this way (The application context that holds all of the services that make up Sakai)

Reply:
I'm not convinced on this one. We used to have a `SakaiApplicationContext` and it was used pretty differently. I'd prefer to use a different name. Also, it's quite likely that, once the dust settles, we do away with this entirely.

It served a helpful purpose during the refactoring, but it has shaken out that the "post processor creator" notion can be handled with the direct spring lifecycle. I've left it in "just in case" for the time being. I think we should merge as-is and, once we have some elapsed time running and developing, revisit to remove this class.

Reply:
sure thing it was a minor suggestion